lib/buffer.js

```diff
@@ -437,15 +439,15 @@ Buffer.concat = function concat(list, length) {
   if (list.length === 0)
     return new FastBuffer();
 
-  if (length === undefined) {
+  if (length == null) {
     length = 0;
     for (i = 0; i < list.length; i++) {
       if (list[i].length) {
         length += list[i].length;
       }
     }
   } else {
-    length = length >>> 0;
+    validateInteger(length, 'length');
```

Comment:
Shouldn’t all of these be `validateUint32`, when it comes to `>>> 0`?

Reply:
The spec itself does not have a limit as far as I know but V8 has a hard limit of 2 ** 31. I'll switch to `validateInt32` and add a test to verify that V8 did not raise the limit.

Reply:
Updated. PTAL